Mathlib/GroupTheory/GroupAction/FixedPoints.lean

```diff
@@ -0,0 +1,292 @@
+  do not commute from the disjointness of the `(fixedBy α g)ᶜ` set and `h • (fixedBy α g)ᶜ`,
+  which is a property used in the proof of Rubin's theorem.
+
+The theorems above are also available for `AddAction`.
+
+## Pointwise group action and `fixedBy (Set α) g`
+
+Since `fixedBy α g = { x | g • x = x }` by definition, properties about the pointwise action of
+a set `s : Set α` can be expressed using `fixedBy (Set α) g`.
+To properly use theorems using `fixedBy (Set α) g`, you should `open Pointwise` in your file.
+
+`s ∈ fixedBy (Set α) g` means that `g • s = s`, which is equivalent to say that
+`∀ x, g • x ∈ s ↔ x ∈ s` (the translation can be done using `MulAction.set_mem_fixedBy_iff`).
+
+`s ∈ fixedBy (Set α) g` is a weaker statement than `s ⊆ fixedBy α g`: the latter requires that
+all points in `s` are fixed by `g`, whereas the former only requires that `g • x ∈ s`.
+-/
+
+namespace MulAction
+open Pointwise
+
+variable {α : Type*}
+variable {G : Type*} [Group G] [MulAction G α]
+variable {M : Type*} [Monoid M] [MulAction M α]
+
+
+section FixedPoints
+
+variable (α) in
+/-- In a multiplicative group action, the points fixed by `g` are also fixed by `g⁻¹` -/
+@[to_additive (attr := simp)
+  "In an additive group action, the points fixed by `g` are also fixed by `g⁻¹`"]
+theorem fixedBy_inv_eq_fixedBy (g : G) : fixedBy α g⁻¹ = fixedBy α g := by
+  ext x
+  rw [mem_fixedBy, mem_fixedBy, inv_smul_eq_iff, eq_comm]
+
+@[to_additive]
+theorem smul_mem_fixedBy_iff_mem_fixedBy {a : α} {g : G} :
+    g • a ∈ fixedBy α g ↔ a ∈ fixedBy α g := by
+  rw [mem_fixedBy, smul_left_cancel_iff]
+  rfl
+
+@[to_additive]
+theorem smul_inv_mem_fixedBy_iff_mem_fixedBy {a : α} {g : G} :
+    g⁻¹ • a ∈ fixedBy α g ↔ a ∈ fixedBy α g := by
+  rw [← fixedBy_inv_eq_fixedBy, smul_mem_fixedBy_iff_mem_fixedBy, fixedBy_inv_eq_fixedBy]
+
+@[to_additive minimalPeriod_eq_one_of_fixedBy]
+theorem minimalPeriod_eq_one_of_fixedBy {a : α} {g : G} (a_in_fixedBy : a ∈ fixedBy α g) :
+    Function.minimalPeriod (fun x => g • x) a = 1 := by
+  rwa [Function.minimalPeriod_eq_one_iff_isFixedPt]
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_subset_fixedBy_zpow (g : G) (j : ℤ) :
+    fixedBy α g ⊆ fixedBy α (g ^ j) := by
+  intro a a_in_fixedBy
+  rw [mem_fixedBy, zpow_smul_eq_iff_minimalPeriod_dvd,
+    minimalPeriod_eq_one_of_fixedBy a_in_fixedBy, Nat.cast_one]
+  exact one_dvd j
+
+variable (M α) in
+@[to_additive (attr := simp)]
+theorem fixedBy_one_eq_univ : fixedBy α (1 : M) = Set.univ :=
+  Set.eq_univ_iff_forall.mpr (fun a => one_smul M a)
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_mul (m₁ m₂ : M) : fixedBy α m₁ ∩ fixedBy α m₂ ⊆ fixedBy α (m₁ * m₂) := by
+  intro a ⟨h₁, h₂⟩
+  rw [mem_fixedBy, mul_smul, h₂, h₁]
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_conj (g h : G) :
+    fixedBy α (h * g * h⁻¹) = (fun a => h⁻¹ • a) ⁻¹' fixedBy α g := by
+  ext a
+  rw [Set.mem_preimage, mem_fixedBy, mem_fixedBy]
```

Comment:
I would suggest just cleaning up the statement of this theorem and deleting one or the other.
```suggestion
theorem fixedBy_conj (g h : G) : fixedBy α (h * g * h⁻¹) = h • fixedBy α g := by
  ext a
  rw [Set.mem_smul_set_iff_inv_smul_mem, mem_fixedBy, mem_fixedBy]
```